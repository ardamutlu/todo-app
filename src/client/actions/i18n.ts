import { State } from '../reducers/i18n';

export type Actions = ReturnType<typeof setLanguage>;

export const SET_LANGUAGE = 'SET_LANGUAGE';

export const setLanguage = (lang: State['lang']) =>
  ({
    type: SET_LANGUAGE,
    payload: {
      lang
    }
  } as const);
