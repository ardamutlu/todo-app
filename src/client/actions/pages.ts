import { State } from '../reducers/pages';

export type Actions =
  | ReturnType<typeof setEnv>
  | ReturnType<typeof setIsMobile>
  | ReturnType<typeof setBaseUrl>
  | ReturnType<typeof resetPageStatus>
  | ReturnType<typeof loadAppProcess>
  | ReturnType<typeof loadAppProcessSuccess>
  | ReturnType<typeof loadAppProcessFailure>
  | ReturnType<typeof loadHomePage>
  | ReturnType<typeof loadHomePageSuccess>
  | ReturnType<typeof loadHomePageFailure>
  | ReturnType<typeof loadDocumentationPage>
  | ReturnType<typeof loadDocumentationPageSuccess>
  | ReturnType<typeof loadDocumentationPageFailure>;

export const SET_ENV = 'SET_ENV';
export const setEnv = (env: State['env']) =>
  ({
    type: SET_ENV,
    payload: {
      env
    }
  } as const);

export const SET_IS_MOBILE = 'SET_IS_MOBILE';
export const setIsMobile = (isMobile: State['isMobile']) =>
  ({
    type: SET_IS_MOBILE,
    payload: {
      isMobile
    }
  } as const);

export const SET_BASE_URL = 'SET_BASE_URL';
export const setBaseUrl = (baseUrl: State['baseUrl']) =>
  ({
    type: SET_BASE_URL,
    payload: {
      baseUrl
    }
  } as const);

export const RESET_PAGES_STATUS = 'RESET_PAGES_STATUS';
export const resetPageStatus = () =>
  ({
    type: RESET_PAGES_STATUS
  } as const);

export const LOAD_APP_PROCESS = 'LOAD_APP_PROCESS';
export const loadAppProcess = () =>
  ({
    type: LOAD_APP_PROCESS
  } as const);

export const LOAD_APP_PROCESS_SUCCESS = 'LOAD_APP_PROCESS_SUCCESS';
export const loadAppProcessSuccess = () =>
  ({
    type: LOAD_APP_PROCESS_SUCCESS
  } as const);

export const LOAD_APP_PROCESS_FAILURE = 'LOAD_APP_PROCESS_FAILURE';
export const loadAppProcessFailure = (err: Error) =>
  ({
    type: LOAD_APP_PROCESS_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_HOME_PAGE = 'LOAD_HOME_PAGE';
export const loadHomePage = () =>
  ({
    type: LOAD_HOME_PAGE
  } as const);

export const LOAD_HOME_PAGE_SUCCESS = 'LOAD_HOME_PAGE_SUCCESS';
export const loadHomePageSuccess = () =>
  ({
    type: LOAD_HOME_PAGE_SUCCESS
  } as const);

export const LOAD_HOME_PAGE_FAILURE = 'LOAD_HOME_PAGE_FAILURE';
export const loadHomePageFailure = (err: Error) =>
  ({
    type: LOAD_HOME_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_DOCUMENTATION_PAGE = 'LOAD_DOCUMENTATION_PAGE';
export const loadDocumentationPage = () =>
  ({
    type: LOAD_DOCUMENTATION_PAGE
  } as const);

export const LOAD_DOCUMENTATION_PAGE_SUCCESS = 'LOAD_DOCUMENTATION_PAGE_SUCCESS';
export const loadDocumentationPageSuccess = () =>
  ({
    type: LOAD_DOCUMENTATION_PAGE_SUCCESS
  } as const);

export const LOAD_DOCUMENTATION_PAGE_FAILURE = 'LOAD_DOCUMENTATION_PAGE_FAILURE';
export const loadDocumentationPageFailure = (err: Error) =>
  ({
    type: LOAD_DOCUMENTATION_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);
