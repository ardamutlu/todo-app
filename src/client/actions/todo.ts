export type Actions =
  | ReturnType<typeof getAllTodoRequest>
  | ReturnType<typeof getAllTodoSuccess>
  | ReturnType<typeof getAllTodoFailure>
  | ReturnType<typeof deleteTodoByIdRequest>
  | ReturnType<typeof deleteTodoByIdSuccess>
  | ReturnType<typeof deleteTodoByIdFailure>
  | ReturnType<typeof updateTodoStatusByIdRequest>
  | ReturnType<typeof updateTodoStatusByIdSuccess>
  | ReturnType<typeof updateTodoStatusByIdFailure>
  | ReturnType<typeof addNewTodoRequest>
  | ReturnType<typeof addNewTodoSuccess>
  | ReturnType<typeof addNewTodoFailure>
  | ReturnType<typeof updateTodoByIdRequest>
  | ReturnType<typeof updateTodoByIdSuccess>
  | ReturnType<typeof updateTodoByIdFailure>
  | ReturnType<typeof filterTodo>
  | ReturnType<typeof searchTodo>;

/**
 * GET ALL TODO
 */
export const GET_ALL_TODO_REQUEST = 'GET_ALL_TODO_REQUEST';
export const getAllTodoRequest = () =>
  ({
    type: GET_ALL_TODO_REQUEST
  } as const);

export const GET_ALL_TODO_SUCCESS = 'GET_ALL_TODO_SUCCESS';
export const getAllTodoSuccess = (payload: any) =>
  ({
    type: GET_ALL_TODO_SUCCESS,
    payload
  } as const);

export const GET_ALL_TODO_FAILURE = 'GET_ALL_TODO_FAILURE';
export const getAllTodoFailure = (err: Error) =>
  ({
    type: GET_ALL_TODO_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * DELETE TODO BY ID
 */
export const DELETE_TODO_BY_ID_REQUEST = 'DELETE_TODO_BY_ID_REQUEST';
export const deleteTodoByIdRequest = (payload: number) =>
  ({
    type: DELETE_TODO_BY_ID_REQUEST,
    payload
  } as const);

export const DELETE_TODO_BY_ID_SUCCESS = 'DELETE_TODO_BY_ID_SUCCESS';
export const deleteTodoByIdSuccess = (payload: any) =>
  ({
    type: DELETE_TODO_BY_ID_SUCCESS,
    payload
  } as const);

export const DELETE_TODO_BY_ID_FAILURE = 'DELETE_TODO_BY_ID_FAILURE';
export const deleteTodoByIdFailure = (err: Error) =>
  ({
    type: DELETE_TODO_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * UPDATE TODO STATUS BY ID
 */
export const UPDATE_TODO_STATUS_BY_ID_REQUEST = 'UPDATE_TODO_STATUS_BY_ID_REQUEST';
export const updateTodoStatusByIdRequest = (payload: any) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_REQUEST,
    payload
  } as const);

export const UPDATE_TODO_STATUS_BY_ID_SUCCESS = 'UPDATE_TODO_STATUS_BY_ID_SUCCESS';
export const updateTodoStatusByIdSuccess = (payload: any) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_SUCCESS,
    payload
  } as const);

export const UPDATE_TODO_STATUS_BY_ID_FAILURE = 'UPDATE_TODO_STATUS_BY_ID_FAILURE';
export const updateTodoStatusByIdFailure = (err: Error) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * ADD NEW TODO
 */
export const ADD_NEW_TODO_REQUEST = 'ADD_NEW_TODO_REQUEST';
export const addNewTodoRequest = (payload: any) =>
  ({
    type: ADD_NEW_TODO_REQUEST,
    payload
  } as const);

export const ADD_NEW_TODO_SUCCESS = 'ADD_NEW_TODO_SUCCESS';
export const addNewTodoSuccess = (payload: any) =>
  ({
    type: ADD_NEW_TODO_SUCCESS,
    payload
  } as const);

export const ADD_NEW_TODO_FAILURE = 'ADD_NEW_TODO_FAILURE';
export const addNewTodoFailure = (err: Error) =>
  ({
    type: ADD_NEW_TODO_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * UPDATE TODO BY ID
 */
export const UPDATE_TODO_BY_ID_REQUEST = 'UPDATE_TODO_BY_ID_REQUEST';
export const updateTodoByIdRequest = (payload: any) =>
  ({
    type: UPDATE_TODO_BY_ID_REQUEST,
    payload
  } as const);

export const UPDATE_TODO_BY_ID_SUCCESS = 'UPDATE_TODO_BY_ID_SUCCESS';
export const updateTodoByIdSuccess = (payload: any) =>
  ({
    type: UPDATE_TODO_BY_ID_SUCCESS,
    payload
  } as const);

export const UPDATE_TODO_BY_ID_FAILURE = 'UPDATE_TODO_BY_ID_FAILURE';
export const updateTodoByIdFailure = (err: Error) =>
  ({
    type: UPDATE_TODO_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * FILTER TODO
 */
export const FILTER_TODO = 'FILTER_TODO';
export const filterTodo = (payload: any) =>
  ({
    type: FILTER_TODO,
    payload
  } as const);

/**
 * SEARCH TODO
 */
export const SEARCH_TODO = 'SEARCH_TODO';
export const searchTodo = (query: string) =>
  ({
    type: SEARCH_TODO,
    payload: {
      query
    }
  } as const);
