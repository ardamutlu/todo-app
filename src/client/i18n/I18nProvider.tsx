import React from 'react';
import { useSelector } from 'react-redux';
import { IntlProvider } from 'react-intl';
import '@formatjs/intl-relativetimeformat/polyfill';
import '@formatjs/intl-relativetimeformat/dist/locale-data/en';
import '@formatjs/intl-relativetimeformat/dist/locale-data/tr';

import enMessages from './messages/en';
import trMessages from './messages/tr';

const allMessages = {
  en: enMessages,
  tr: trMessages
};

export default function I18nProvider({ children }: any) {
  const locale = useSelector(({ i18n }: any) => i18n.lang);
  const messages = allMessages[locale];

  return (
    <IntlProvider locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
}
