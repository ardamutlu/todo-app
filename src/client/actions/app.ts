import { State } from '../reducers/pages';

export type Actions = ReturnType<typeof setIsMobile>;

export const SET_IS_MOBILE = 'SET_IS_MOBILE';
export const setIsMobile = (isMobile: State['isMobile']) =>
  ({
    type: SET_IS_MOBILE,
    payload: {
      isMobile
    }
  } as const);
