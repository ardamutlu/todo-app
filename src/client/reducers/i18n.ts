import * as Actions from '../actions/i18n';

export type State = {
  lang: string;
};

export const initialState: State = {
  lang: 'en'
};

export const reducer = (state: State = initialState, action: Actions.Actions): State => {
  switch (action.type) {
    case Actions.SET_LANGUAGE:
      localStorage.setItem('lang', action.payload.lang);
      return { ...state, lang: action.payload.lang };

    default:
      return state;
  }
};
