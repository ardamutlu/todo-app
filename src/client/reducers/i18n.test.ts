import { reducer, initialState } from './i18n';
import * as actions from '../actions/i18n';

test('should return the initial state', () => {
  expect(reducer(undefined, { type: undefined } as any)).toEqual(initialState);
});

test('should handle SET_LANGUAGE', () => {
  expect(reducer(initialState, actions.setLanguage('en'))).toMatchSnapshot();
});
