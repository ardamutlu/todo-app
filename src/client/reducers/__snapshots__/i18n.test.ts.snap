// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle SET_LANGUAGE 1`] = `
Object {
  "lang": "en",
}
`;
